Formula/f/firefly.rb

```diff
@@ -0,0 +1,38 @@
+class Firefly < Formula
+  desc "Command-line Interface (CLI) for Hyperledger FireFly"
+  homepage "https://hyperledger.github.io/firefly/latest/"
+  url "https://github.com/hyperledger/firefly-cli/archive/refs/tags/v1.3.1.tar.gz"
+  sha256 "d2b0420b37c1ce6195e0739b2341502e65fea23c3ddd41cd55159ea237e01f23"
+  license "Apache-2.0"
+  head "https://github.com/hyperledger/firefly-cli.git", branch: "main"
+
+  livecheck do
+    url :stable
+    # This regex is intended to match FireFly version tags (e.g. `v1.2.3`)
+    # and omit any pre-release suffixes (e.g. `v1.2.3-rc0`)
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    build_commit = if build.head?
+      `git rev-parse --short HEAD`.chomp
+    else
+      "unknown"
+    end
+
+    ldflags = %W[
+      -X github.com/hyperledger/firefly-cli/cmd.BuildDate=#{Time.now.utc.iso8601}
+      -X github.com/hyperledger/firefly-cli/cmd.BuildCommit=#{build_commit}
+      -X github.com/hyperledger/firefly-cli/cmd.BuildVersionOverride=v#{version}
+    ]
+    system "go", "build", *std_go_args(ldflags:), "ff/main.go"
+
+    generate_completions_from_executable(bin/"firefly", "completion")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/firefly version --short")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This CLI tool is a bit unique in that it primarily operates through an interactive command, which complicates creating non-interactive tests as outlined in the formula cookbook. Given the nature of the tool, simulating the full functionality would be challenging in this context.

However, I’ve added a test case that checks if the engine is not running, which might be analogous to the scenario where a CLI requires an external dependency like Docker. I believe this approach provides a reasonable compromise while still verifying some core functionality of the tool.

Reply:
I think we could make this non-interactive and test with the `ff init` command that is supposed to lay down files.